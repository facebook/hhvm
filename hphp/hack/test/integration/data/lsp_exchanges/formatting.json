[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":{}
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/formatting",
  "id": 2,
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
      },
      "options": {"tabSize": 5, "insertSpaces": true}
   }
},
{
  "jsonrpc": "2.0",
  "id": 3,
  "method": "shutdown",
  "params": {}
}
]
