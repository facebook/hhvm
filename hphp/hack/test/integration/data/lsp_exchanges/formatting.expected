[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": [
      {
        "range": {
          "start": {
            "line": 0,
            "character": 0
          },
          "end": {
            "line": 15,
            "character": 0
          }
        },
        "newText": "<?hh //strict\n\nfunction x(): string {\n     /* @lint-ignore TXT2 3 tabs on purpose */\n     $a = \"this\";\n\n     /* @lint-ignore TXT2 2 tabs on purpose */\n     $b = \"is\";\n\n     /* lint-ignore TXT2 1 tab on purpose */\n     $c = \"messy\"; // 1 tab\n\n     $d = \".\"; // 4 spaces\n     return \"$a\".\"$b\".\"$c\".\"d\";\n}\n"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": null
  }
]